### ISSUE TITLE
lab-kubernetes-challenge-1 Pod fails to Validate

### ISSUE THREAD

**Issue by ioideanh** [kodekloudhub/kubernetes-challenges#11]
lab-kubernetes-challenge-1
You can not validate the Pod part,  the command always fails to validate. 
After attempting this a couple of times and failing on the command, I copied and pasted your pod yaml and it still fails to validate

**Comment by AbhishekHyd123** [comment]
same here. The pod is not getting scheduled on any node. It is evident from this thread.
https://kodekloud.slack.com/archives/C02LS58EGQ4/p1693683541377539
See all the snapshots given by Abhishek-G
The error says out of 9 points mentioned for pod, below one is not completed correctly.

initContainer: 'copy-jekyll-site', command: [ "jekyll", "new", "/site" ] (command to run: jekyll new /site) 

Can someone please work on it ? Also notify us when the root cause is identified.

**Comment by masteryyh** [comment]
still the same here

**Comment by panromir** [comment]
Just ran into this myself, issue persists. :(

**Comment by al3x609** [comment]
_my troubleshoot path_.

**__What I show below is just a possible cause of the error, and its solution__**

short fix > 
- set __storageClassName: local-storage__ on PVC manifest and try again.
____
TL;DR
1. for ie. open JS inspector console in Chrome , go to console tab.
2. then, click on check button, get the XHR response. 16th json object. 
real command without the SSH login
`kubectl --kubeconfig /etc/kubernetes/admin.conf -n development logs  jekyll -c copy-jekyll-site | tail -1 | awk -F / '{print $NF}'`

 ![Screen Shot 2024-04-10 at 1 55 00](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/4e341a0e-2f76-4ae9-af02-652d61267bc7)


3. in a sucessfull response, the command return __site__ word with some hex unicode chars
4. you can test the command directy over the k8s cluster. 
check de exit code from previews cmd with __$?__
![Screen Shot 2024-04-10 at 2 15 19](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/d12eca70-52bf-404d-9d50-8e7c9b2b432e)

- [ x]  _his was only for thes backend and frontend  Web interaction OK. until now._

____________


What was the response in failed case ?
 
 _Something different from zero and without any text in STD output to compare with._
 
 ____
0. check the PV status
 ![Screen Shot 2024-04-10 at 2 38 43](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/fc18849a-3486-4a36-9f34-555312f72b9d) , before that nothing.

1.  Now, if, between tests, you delete the PVC, an error is generated with PV plugins, indicating that you need recreate the PV for a manual release

![Screen Shot 2024-04-10 at 2 51 30](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/6c81cfcc-1cbe-47b5-a12c-923ff3705772)


because there is not an available StorageClass for automatic thasks (__no provisioner__)
![Screen Shot 2024-04-10 at 4 18 30](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/c1a42413-5753-4948-96de-d8537d7f5ae8)


2. Now if you forgot set the field storage class on the PVC manifest,  this will trigger an error.

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: jekyll-site
  namespace: development
spec:
  volumeName: jekyll-site
  accessModes:
    - ReadWriteMany
  #storageClassName: local-storage
  resources:
    requests:
      storage: 1Gi
```



 ![Screen Shot 2024-04-10 at 4 35 30](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/c3973067-63a4-4ed2-9b75-01ee08549fd5)


__but!!__ 
__the test for PVC status as the fist part show ... Everything is going well with the PVC.
![Screen Shot 2024-04-10 at 4 47 43](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/af0c1747-c261-487d-8073-6ca066da2282)

And the deployment of the POD shows us...

![Screen Shot 2024-04-10 at 4 51 45](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/1560626c-e9e5-43dd-85a3-f9890b00dd0a)

_____
 
 finaly with the approach ...
 
![Screen Shot 2024-04-10 at 5 54 28](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/688c125d-7e67-4291-8def-f9708971cbad)


![Screen Shot 2024-04-10 at 6 15 52](https://github.com/kodekloudhub/kubernetes-challenges/assets/4086644/1b6ab407-772b-4b4f-a558-1aa4ac0106c8)

**Comment by BitterDone** [comment]
I'm also having this issue, exactly as described numerous times here:
https://kodekloud.com/community/t/kodekloud-challenge-1-issue-with-creating-the-pod-issue-with-command/131666

I've reset the lab about 4 times. I've diff'd my yaml files against those in the Github solutions and mine were the same. Even after pasting the solution yaml files and commands, it doesn't validate on the `command: [ "jekyll", "new", "/site" ]` step.

**Comment by AIWithShrey** [comment]
Still no official fix for this problem? Running into this myself now. When I look at the logs of the pod, here's what I see:

```
kubectl logs -n development jekyll 
Defaulted container "jekyll" out of: jekyll, copy-jekyll-site (init)
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
Fetching gem metadata from https://rubygems.org/............
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
ffi-1.17.0-x86_64-linux-musl requires rubygems version >= 3.3.22, which is
incompatible with the current version, 3.0.3
```

**Comment by fireflycons** [comment]
We have merged https://github.com/kodekloudhub/kubernetes-challenges/pull/13
Please try the challenge again

@AIWithShrey @BitterDone
